### DISCUSSION TITLE
Get the current file's read time (i.e. "How long to read")

### DISCUSSION THREAD

**Discussion by PuppyGirlBelly** [SilentVoid13/Templater#1123] [Templates Showcase]
Hi, I have spent way too much time ~~procrastinating~~ working on this; so I figured I should try and share so somebody else doesn't have to spend this effort.

I like to have a part of my frontmatter to list how long a page is; as I use Obsidian to hold documents I want to read offline. So I created a template that should give an approximate estimate of how long a document is. It's not the most accurate as `tp.file.content` still includes frontmatter; but I have filtered out most non-readable characters so it should give a decent ballpark.

<details><summary>Version 1.0</summary>

- [Gist of the script](https://gist.github.com/SoftAnnaLee/1bc70c7e25717e5e6e3b6d2b19a4fea6)
  - Includes code that outputs in a 'human readable' format, for example;
    - 128 minutes reads as "2 hours and 8 minutes"
    - 60 minutes reads as "1 hour"
    - sub-1 minute reads as "Less than one minute"

- Inline Template 
``` javascript
<%*
    const WPM = 255 
    const words = await tp.file.content.trim().replace(/[^\w\s]/g, "").split(/\s+/).filter((x) => x !== "");
    const readTime = Math.ceil(words.length / WPM);
%>
read time: <% readTime %> minutes
```
</details>

Let me know of any improvements I can make or any additions folks might need!

**Comment by Dercraker** [comment]
I made a small Upgrade. The template converts to Hours and Minutes if over 60 Minutes
```js
<%*
const WPM = 255 
const words = await tp.file.content.trim().replace(/[^\w\s]/g, "").split(/\s+/).filter((x) => x !== "");
const readTime = Math.ceil(words.length / WPM);
const totalHours = (readTime / 60);
const readHours = Math.floor(totalHours);
const totalMinutes = (totalHours - readHours) * 60;
const readMinutes = Math.round(totalMinutes);

if (readHours != 0)
	tR += `${readHours} hour${readHours == 1 ? "": "s"} and `
tR += `${readMinutes} minute${readMinutes == 1 ? "": "s"}`
%>
```

**Comment by SupremeToaster** [comment]
That's super awesome! Thanks for sharing!

**Comment by FroguelYipit** [comment]
Is there a way to auto-update this every time I open the doc? Maybe if this is turned on a file property? Many documents get increased over time and this would be super useful.

**Reply by Gryn23** [reply]
This is so important. At the moment the template only spits out the number of minutes it takes to read the note when created. Which is typically 0 minutes, as you still need to write the content. 

Anyone know a way to update this script when the file is updated?

**Reply by Zachatoo** [reply]
If you need it to dynamically update, I'd recommend using the Dataview plugin instead. Here's an example of a "how long to read" dynamic query I made a while ago.
https://github.com/SilentVoid13/Templater/issues/913#issuecomment-1711104067

**Reply by Gryn23** [reply]
Thanks a lot @Zachatoo !
